spaceapi_server/src/lib.rs

```diff
@@ -94,32 +88,118 @@ impl SpaceapiServer {
         Iron::new(self).http((host, port)).unwrap();
     }
 
-}
+    /// Register a new sensor.
+    ///
+    /// The first argument is a ``spaceapi::Sensor`` instance containing all static data. The
+    /// second argument specifies how to get the actual sensor value from the datastore. And the
+    /// third argument specifies the data type of the value.
+    pub fn register_sensor(&mut self, sensor: spaceapi::SensorTemplate, data_key: String, data_type: SensorValueType) {
+        self.sensors.push(
+            SensorSpec { sensor: sensor, data_key: data_key, data_type: data_type }
+        );
+    }
 
-impl middleware::Handler for SpaceapiServer {
+    fn build_response_json(&self) -> Json {
```

Comment:
Yes, yes, it definitely needs refactoring. You could do that :grin: But it works right now, so it's a start.